### PR TITLE
Implemented __reduce__ method for the ImportError exception

### PR DESCRIPTION
**Pull request by LucaSforza**
Added `__reduce__` method for `ImportError` #3611.

### CODE REVIEW

**Comment by fanninpm** [comment]
cc @Snowapril

File: vm/src/exceptions.rs [hunk 1]
```diff
@@ -1182,6 +1182,20 @@ pub(super) mod types {
             )?;
             Ok(())
         }
+        #[pymethod(magic)]
+        fn reduce(exc: PyBaseExceptionRef, _vm: &VirtualMachine) -> PyTupleRef {
```
Comment (youknowone) [inline vm/src/exceptions.rs:1186]:
```suggestion
        fn reduce(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyTupleRef {
```
`_` means unused variable. But it is used below

File: vm/src/exceptions.rs [hunk 2]
```diff
@@ -1182,6 +1182,20 @@ pub(super) mod types {
...
+            let obj = exc.as_object().to_owned();
+            let mut result: Vec<PyObjectRef> = vec![
+                obj.class().to_owned().into(),
+                _vm.new_tuple((exc.get_arg(0).unwrap(),)).into(),
```
Comment (youknowone) [inline vm/src/exceptions.rs:1190]:
```suggestion
                vm.new_tuple((exc.get_arg(0).unwrap(),)).into(),
```

File: vm/src/exceptions.rs [hunk 3]
```diff
@@ -1182,6 +1182,20 @@ pub(super) mod types {
...
+            ];
+
+            if let Some(dict) = obj.dict().filter(|x| !x.is_empty()) {
+                result.push(dict.into());
+            }
+
+            result.into_pytuple(_vm)
```
Comment (youknowone) [inline vm/src/exceptions.rs:1197]:
```suggestion
            result.into_pytuple(vm)
```

**Review by youknowone — APPROVED** [review]
Thank you for contributing!

**Pull request merged**